### DISCUSSION TITLE
Local Area Network

### DISCUSSION THREAD

**Discussion by axmad386** [NativePHP/laravel#381] [Q&A] [answered]
Hello, Is there any way to create local connection via LAN. For example I have two desktop application created by native php and can communication each other. Or may be I miss read the documentation

**Comment by simonhamp** [comment]
The networking features of PHP just sit on top of the operating system's network stack, so you should have no problems making requests across the local network from PHP.

As for getting one app to talk to another across the network, you'd need to have the app spin up some kind of socket service that is able to listen on an available port and on the correct IP address.

I recommend you prototype this outside of NativePHP to start with (e.g. just write a PHP script that does this), to prove out the logic and figure out some of your edge cases before trying to add this complexity inside your NativePHP app.

Some things you'll need to consider:
- If you expect this process to be long-lived (e.g. booting up when your app opens and shutting down when the app quits), you'll need a way to manage the process in case it crashes, so that you're able to spin it up again - we're already working on something that may help with this
- You may need to think about firewalls to make sure the ports you want to use on each machine are open for two-way communication
- The address mechanism used will depend largely on your network's configuration - can you used name-based addressing? Or will you be dependent on IP addresses? How will you get each app to be aware of the other machine's address?
- Security will be an important consideration too - how do you make sure that it's only your app that's able to use this comms channel? As well as deciding which functions are exposed

None of this is peculiar to NativePHP.

**Reply by axmad386** [reply]
Oh, maybe I'll change my question a little bit. Can the API routes provided by Laravel be accessed from outside a NativePHP app?

In this documentation [NativePHP Docs](https://nativephp.com/docs/1/digging-deeper/security#the-web-servers), you mention that NativePHP spins up two web servers: the Application server and the Electron server. However, I don't see where the Application server is running, or on what port, for instance.

**Reply by simonhamp** [reply, marked as answer]
> Can the API routes provided by Laravel be accessed from outside a NativePHP app?

No. You'd need to spin up a separate process to create your own service.

All of my considerations above would still apply.

We are [working on](https://github.com/orgs/NativePHP/discussions/371) a standardised approach to this.

> you mention that NativePHP spins up two web servers

You should not rely on the fact that NativePHP _currently_ uses web servers internally - this behaviour is very likely to change in the near future.

**Reply by axmad386** [reply]
Thanks for the explanation sir

**Comment by simonhamp** [comment]
Our Child Processes functionality [got released](https://github.com/NativePHP/laravel/releases/tag/0.6.0) yesterday!

You'd now just need to start a process on each side that listens on a port that is accessible to other devices on the network - for example, you may be able to use `netcat` or `telnet` to achieve a basic level of sending messages from your app across the network to other devices running your app.
